Mathlib/Algebra/Order/AbsoluteValue/Nonarchimedean.lean

```diff
@@ -0,0 +1,95 @@
+/-
+Copyright (c) 2025 Fabrizio Barroero. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Fabrizio Barroero
+-/
+import Mathlib.Algebra.BigOperators.Group.Finset.Basic
+import Mathlib.Algebra.Order.AbsoluteValue.Basic
+
+/-!
+# Nonarchimedean absolute values
+
+An absolute value `f` is nonarchimedean if it satisfies `f (x + y) ≤ max (f x) (f y)` for all
+`x` and `y` in `R`.
+-/
+
+variable {α R S : Type*}
+
+namespace AbsoluteValue
+
+section LinearOrderedSemiring
+
+variable [LinearOrderedSemiring S]
+
+/-- An absolute value `f` is nonarchimedean if it satisfies `f (x + y) ≤ max (f x) (f y)` for all
+`x` and `y` in `R`. -/
+def IsNonarchimedean [Semiring R] (f : AbsoluteValue R S) : Prop :=
+  ∀ x y : R, f (x + y) ≤ max (f x) (f y)
+
+open Finset in
+/-- Ultrametric inequality with `Finset.Sum`. -/
+lemma apply_sum_le_sup_of_isNonarchimedean [Semiring R] {f : AbsoluteValue R S}
```

Comment:
```suggestion
lemma apply_sum_le_sup_of_isNonarchimedean [AddCommMonoid R] {f : R → S}
```
If you follow my suggestion about the `IsNonarchimedean` definition, you can relax these hypotheses.